.github/workflows/ci-fork.yml

```diff
@@ -0,0 +1,62 @@
+name: CI (Fork)
+
+on:
+  workflow_run:
+    workflows: ["CI"]
+    types:
+      - completed
+
+jobs:
+  unit-test-results:
+    name: Unit Test Results
+    runs-on: ubuntu-latest
+    if: >
+      github.event.workflow_run.conclusion != 'skipped' &&
+      github.event.workflow_run.head_repository.full_name != github.repository
+
+    steps:
+      - name: Download Artifacts
+        uses: actions/github-script@v3
+        with:
+          script: |
+            var fs = require('fs');
+            var path = require('path');
+            var artifacts_path = path.join('${{github.workspace}}', 'artifacts')
+            fs.mkdirSync(artifacts_path, { recursive: true })
+
+            var artifacts = await github.actions.listWorkflowRunArtifacts({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: ${{ github.event.workflow_run.id }},
+            });
+
+            for (const artifact of artifacts.data.artifacts) {
+               var download = await github.actions.downloadArtifact({
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  artifact_id: artifact.id,
+                  archive_format: 'zip',
+               });
+               var artifact_path = path.join(artifacts_path, `${artifact.name}.zip`)
+               fs.writeFileSync(artifact_path, Buffer.from(download.data));
+               console.log(`Downloaded ${artifact_path}`);
+            }
+      - name: Extract Artifacts
+        run: |
+          for file in artifacts/*.zip
+          do
+            if [ -f "$file" ]
+            then
+              dir="${file/%.zip/}"
+              mkdir -p "$dir"
+              unzip -d "$dir" "$file"
+            fi
+          done
+
+      - name: Publish unit test results
+        uses: EnricoMi/publish-unit-test-result-action@v1
+        with:
+          check_name: unit-test-results
+          comment_title: Unit Test Results
+          files: "artifacts/*/**/*.xml"
+          commit: ${{ github.event.workflow_run.head_sha }}
```

Comment:
Does your action add a comment to the linked PR when only a commit is specified?

Reply:
The `commit` option is needed here because the `workflow_run` event's `GITHUB_SHA` is master HEAD, whereas the previous `CI` workflow had a different commit SHA. The default for `commit` is `GITHUB_SHA`. The previous `CI`workflow event's `GITHUB_SHA` is `github.event.workflow_run.head_sha`.

Reply:
Sure, but does your action then lookup the relevant PR for publishing the comments?

Reply:
Nevermind, I had a look at your code and it does 👌

Reply:
yes, that's the purpose of providing the right `commit` here